src/pages/VenuePage/VenuePage.tsx

```diff
@@ -173,20 +263,44 @@ export const VenuePage: React.FC = () => {
 
   // useVenueAccess(venue, handleAccessDenied);
 
-  if (venueRequestStatus && !venue) {
+  if (isVenueLoaded && !venue) {
     return <>This venue does not exist</>;
   }
 
-  if (!venue || !venueId) {
-    return <LoadingPage />;
+  if (!venueId) {
+    // this should be happening only if invalid url param
+    // no sense in displaying loading page, another message or action is needed
+    // @debt use/display proper message/action when !venueId
+    return null;
+  }
+
+  if (!venue) {
+    return (
+      <Suspense fallback={<></>}>
+        <LoadingPage />
+      </Suspense>
+    );
   }
 
   if (!user) {
-    return <Login venue={venue} />;
+    return (
+      <>
+        <Preload venue={venue} />
+        <Login venue={venue} />
+      </>
+    );
   }
 
   if (!profile) {
-    return <LoadingPage />;
+    return (
+      <>
+        <Preload venue={venue} />
+        {
+          // is it really necessary to display loading page here?
+          // <LoadingPage />
+        }
+      </>
+    );
```

Comment:
At this stage i'm not convinced that `Preload` is going to have any meaningful impact on things; and if we do decide to keep it, it shouldn't end up duplicated in so many places.

Reply:
Replaced with hook in e9da1d5788e915f6de13a7aa0315601017c75b73 . Preloading does show impact in the _Cumulative Layout Shift_ metric at order of a magnitude:

- with `usePreloadAssets()` hook, cumulative layout shift is in the range of **0.01** and **0.02** (good)
- without the hook, cumulative layout shift is above **0.1** (not good, not terrible)

Reply:
This is likely just because the map becomes available earlier, which we can solve better by properly setting the expected width/height on it ahead of time.

OOS of this PR though.

Reply:
The map and rooms are the first things user interacts with so loading those images sooner helps , but the most notable difference is the cumulative layout shift, it goes above the threshold of 0.1 seconds (the next one is 0.25)